src/lib/auth.ts

```diff
@@ -0,0 +1,42 @@
+import { atom, selector, setState } from "atom.io"
+import * as FirebaseAuth from "firebase/auth"
+
+import type { SystemUser } from "../types"
+import { auth } from "./firebase"
+import { systemUserAtoms } from "./users"
+
+export type AuthStatus =
+	| { authenticated: false; loading: boolean }
+	| { authenticated: true; loading: false; me: FirebaseAuth.User }
```

Comment:
nice type!

Reply:
Though I'm not necessarily onboard with `me` and `myself` — seems like you should pick a lane here? Also not yet convinced this is better than `currentUser`